File: FirstApp/Stock/UserControl/AttributeControl.vb
Public Class AttributeControl

    Private _currentAttribute As Item.Item_Attribute

    Public Property CurrentAttribute As Item.Item_Attribute
        Get
            Return _currentAttribute
        End Get
        Set(value As Item.Item_Attribute)
            _currentAttribute = value
            With _currentAttribute
                AttrIconBT.Text = $"{{{ .DataType}{If(.IsCurrency, "-₹", "")}}} { .Name}" + If(.IsOptional, "*", "")
                If .IsConstant Then
                    AttrIconBT.Font = New Font(AttrIconBT.Font, FontStyle.Underline)
                    Label1.Text = "Fix. Val:"
                End If
                Size = New Point(Size.Width + TextRenderer.MeasureText(AttrIconBT.Text, AttrIconBT.Font).Width + 2, Height)
                If .IsFormula Then
                    DefValueLB.Text = "{RULE}"
                ElseIf .DataType = Item.DataType.List_Type Then
                    DefValueLB.Text = .DefaultValue.Split("`")(0)
                Else
                    DefValueLB.Text = .DefaultValue
                End If

                AttrIconBT.IconChar = .Icon
            End With

        End Set
    End Property

    Public Event ButtonClick(_property As Item.Item_Attribute)
    Private Sub UsedButton_Click(sender As Object, e As EventArgs) Handles UsedButton.Click
        RaiseEvent ButtonClick(_currentAttribute)
    End Sub

End Class

File: OrnaTrack/Stock/UserControl/AttributeControl.cs
using System;
using System.Drawing;
using System.Windows.Forms;

namespace FirstApp
{
    public partial class AttributeControl
    {

        private Item.Item_Attribute _currentAttribute;

        public Item.Item_Attribute CurrentAttribute
        {
            get
            {
                return _currentAttribute;
            }
            set
            {
                _currentAttribute = value;
                {
                    ref var withBlock = ref _currentAttribute;
                    AttrIconBT.Text = $"{{{withBlock.DataType}{(withBlock.IsCurrency ? "-₹" : "")}}} {withBlock.Name}" + (withBlock.IsOptional ? "*" : "");
                    if (withBlock.IsConstant)
                    {
                        AttrIconBT.Font = new Font(AttrIconBT.Font, FontStyle.Underline);
                    }
                    Size = (Size)new Point(Size.Width + TextRenderer.MeasureText(AttrIconBT.Text, AttrIconBT.Font).Width + 2, Height);
                    if (withBlock.IsFormula)
                    {
                        DefValueLB.Text = "{RULE}";
                    }
                    else if ((withBlock.DataType ?? "") == Item.DataType.List_Type)
                    {
                        DefValueLB.Text = withBlock.DefaultValue.Split('`')[0];
                    }
                    else
                    {
                        DefValueLB.Text = withBlock.DefaultValue;
                    }

                    AttrIconBT.IconChar = withBlock.Icon;
                }

            }
        }

        public event ButtonClickEventHandler ButtonClick;

        public delegate void ButtonClickEventHandler(Item.Item_Attribute _property);

        public AttributeControl()
        {
            InitializeComponent();
        }
        private void UsedButton_Click(object sender, EventArgs e)
        {
            ButtonClick?.Invoke(_currentAttribute);
        }

    }
}